es6.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+module.exports = {
+  extends: ['eslint-config-mm/es5'],
+  parserOptions: {
+    ecmaVersion: 6
+    // Can't use ES6 imports yet in node: https://github.com/nodejs/help/issues/53#issuecomment-165763446
+    // sourceType: 'module'
+  },
+  env: {
+    node: true,
+    es6: true
+  },
+  rules: {
+    'array-callback-return': 1,
+    'arrow-parens': [1, 'always'],
+    'arrow-body-style': [1, 'always'],
+    'arrow-spacing': 1,
+    'no-const-assign': 1,
+    'no-iterator': 1,
+    'no-var': 1,
+    'object-shorthand': 1,
+    'prefer-arrow-callback': 1,
+    'prefer-const': 1,
```

Comment:
Nothing in here will imply immutability. For that we would need to move to Purescript or something. However, preferring `const` over `let` is a good smell test to at least keep variable reassignment to a minimum.

The idea here is indeed to have it complain a lot, but to not break CI (which is why the ES6 stuff is all warnings, not errors). We can run it like this for a month and take a little time to clean up every file when we touch it (since most changes are pretty trivial). Maybe we can even let a transpiler do some of the work for us. Then afterwards we plan some time to update the remaining untouched stuff and finally promote these rules to actual errors.